### ISSUE TITLE
Make http.send accept any json variant as application/json

### ISSUE THREAD

**Issue by mrvanes** [open-policy-agent/opa#6684]
## What is the underlying problem you're trying to solve?
Our SCIM server returns `application/scim+json` as a `Content-Type` header. We need to enable `force_json_decode` to get the `raw_body` parsed into `body`.

## Describe the ideal solution
It would be nice if opa would understand that `application/variant+json` is just a variant of `application/json` for any `variant` of json so it would not be necessary to add `force_json_decode` to the request body.

## Describe a "Good Enough" solution
```
users_request := {
    "url": "https://scimserver/Users",
    "method": "GET",
    "headers": {
        "x-api-key": "secret",
        "accept": "application/scim+json"
    }
}
```
... would return a valid decoded body in the result.

**Comment by anderseknert** [comment]
Sounds like a reasonable improvement to me. If you'd like to submit a PR, that'd be neat 🙂

**Comment by srenatus** [comment]
As a workaround, have you tried setting `"force_json_decode": true`?
> Decode the HTTP response message body as JSON even if the Content-Type header is missing. Default: false.
([source](https://www.openpolicyagent.org/docs/latest/policy-reference/#http))

**Comment by srenatus** [comment]
"For more information, please re-read." -- I've removed my comment re force_json_decode, you're already aware. 😅

**Comment by srenatus** [comment]
💭 Right now, application/json is json-decoded, and application/yaml + application/x-yaml are yaml-decoded. So if we go with
```rego
users_request := {
    "url": "https://scimserver/Users",
    "method": "GET",
    "headers": {
        "x-api-key": "secret",
        "accept": "application/scim+json"
    }
}
```
would this mean it should do JSON? How would I declare that a certain extra header value means that it should be treated as YAML?

**Comment by mrvanes** [comment]
No, my suggestion was not to use the "accept" header, that's just to show a correct SCIM request body. My suggestion is to be lenient about the returned `application/<anything+>json` Content-Type.

**Comment by srenatus** [comment]
Ah! Thanks for clarifying. Yeah, that makes this a bit simpler.

**Comment by ashutosh-narkar** [comment]
[This](https://github.com/open-policy-agent/opa/blob/main/topdown/http.go#L1335) method handles the response processing which would need to be updated I think. Feel free to submit a PR if you'd like. Thanks.
